### ISSUE TITLE
Some tests with GRU and CudaDNNGRU

### ISSUE THREAD

**Issue by wirtsi** [pinae/LSTM-Classification#2]
So came across [Cuda based RNN in Keras](https://keras.io/layers/recurrent/#cudnnlstm) and played around with them a bit. A got exceptional good results with CuDNNGRU.

It's more or less a drop-in replacement, you have to drop the recurrent_dropout and dropout keys in the RNN layers.

With a batch size of 1200 on a Tesla P100 (2 Core VM in GCE), an epoch takes around 20s, results after 150 epochs (with other params unchanged) were:

`422us/step - loss: 0.0240 - binary_accuracy: 0.9920 - c_score: 0.5439 - val_loss: 0.0409 - val_binary_accuracy: 0.9884 - val_c_score: 0.3422
INFO - LSTM_Classification - Completed after 0:52:00
Accuracy: 98.84%`

Running the same thing with the normal GRU (then again with the dropout layers) I get

`105s 2ms/step - loss: 0.0274 - binary_accuracy: 0.9907 - c_score: 0.4703 - val_loss: 0.0331 - val_binary_accuracy: 0.9895 - val_c_score: 0.4074`

So what's interesting is that GRU appears to outperform LSTM in terms of c_score. Why not using any dropouts in the Cuda version actually improves the score significantly ... no idea 😕
